code/game/objects/items/weapons/rpd.dm

```diff
@@ -0,0 +1,303 @@
+
+/obj/item/weapon/rpd/ui_interact(mob/user, ui_key = "main", var/datum/nanoui/ui = null, var/force_open = 1, var/datum/topic_state/state = inventory_state)
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, force_open)
+	if(!ui)
+		ui = new(user, src, ui_key, "rpd.tmpl", "[name]", 600, 650, state = state)
+		ui.open()
+		if(!initialised)
+			Initialiseicons()
+		ui.set_auto_update(1)
+
+/obj/item/weapon/rpd/ui_data(mob/user, ui_key = "main", datum/topic_state/state = inventory_state)
+	var/data[0]
+	data["dpipelist"] = dpipelist
+	data["iconrotation"] = iconrotation
+	data["mode"] = mode
+	data["pipelist"] = pipelist
+	data["pipemenu"] = pipemenu
+	data["pipetype"] = pipetype
+	data["whatdpipe"] = whatdpipe
+	data["whatpipe"] = whatpipe
+	data["mainmenu"] = list(
+	list("key1" = "Atmospherics piping", "key2" = "1", "key3" = "wrench"),
+	list("key1" = "Disposals piping", "key2" = "2", "key3" = "recycle"),
+	list("key1" = "Rotate mode", "key2" = "3", "key3" = "rotate-right"),
+	list("key1" = "Flip mode", "key2" = "4", "key3" = "exchange"),
+	list("key1" = "Recycle mode", "key2" = "5", "key3" = "trash"))
+	data["mainmenu"] = list(
+	list("key1" = "Atmospherics piping", "key2" = "1", "key3" = "wrench"),
+	list("key1" = "Disposals piping", "key2" = "2", "key3" = "recycle"),
+	list("key1" = "Rotate mode", "key2" = "3", "key3" = "rotate-right"),
+	list("key1" = "Flip mode", "key2" = "4", "key3" = "exchange"),
+	list("key1" = "Recycle mode", "key2" = "5", "key3" = "trash"))
+	data["pipemenu"] = list(
+	list("key1" = "Normal pipes", "key2" = "1"),
+	list("key1" = "Supply pipes", "key2" = "2"),
+	list("key1" = "Scrubber pipes", "key2" = "3"),
+	list("key1" = "Devices", "key2" = "4"),
+	list("key1" = "HE pipes", "key2" = "5"))
+	return data
+
+/obj/item/weapon/rpd/Topic(href, href_list, nowindow, state)
+	if(href_list["iconrotation"])
+		iconrotation = text2num(href_list["iconrotation"])
+	else if(href_list["whatpipe"])
+		whatpipe = text2num(href_list["whatpipe"])
+	else if(href_list["whatdpipe"])
+		whatdpipe = text2num(href_list["whatdpipe"])
+	else if(href_list["pipetype"])
+		pipetype = text2num(href_list["pipetype"])
+	else if(href_list["mode"])
+		mode = text2num(href_list["mode"])
+	else if(href_list["refresh"])
+		to_chat(usr, "<span class = 'notice'>You press the reset button, refreshing all settings on the [src].</span>")
+		iconrotation = 0
+		mode = 1
+		pipemenu = 1
+		pipetype = 1
+		whatpipe = 0
+		whatdpipe = 0
+		cooldown = 0
+		Initialiseicons()
+	else
+		return
+	nanomanager.update_uis(src)
+	return
+
+//What the RPD actually does
+
+/obj/item/weapon/rpd/afterattack(atom/target, mob/user as mob, proximity)
+	var/turf/T = get_turf(target)
+	if(!proximity || cooldown == 1)
+		return
+	else if(!isturf(target) && !istype(target, /obj/item/pipe) && !istype(target, /obj/item/pipe_meter) && !istype(target, /obj/item/pipe_gsensor) && !istype(target, /obj/structure/disposalconstruct) || istype(target, /turf/simulated/shuttle))
```

Comment:
Yeah but I WANT to be able to hit a status display so I don't have to go pixel hunting.

And putting on table would already happen because it's after attack and not pre attack.